src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -414,6 +455,15 @@ public Response get(String json) throws Exception
 
 			for (QueryMetric query : queries)
 			{
+				Map<String, Collection<String>> tags = query.getTags().asMap();
+				Set<String> keys = tags.keySet();
+
+				for (String key : keys) {
+					Iterator<String> iter = tags.get(key).iterator();
+					while(iter.hasNext())
+						span.setTag(key, iter.next());
+				}
```

Comment:
In theory this will work for single query, but for multiple queries we might overwrite span tags. I haven't checked the underlying layer of ``DatastoreQuery``, but I feel it is a better place for setting those tags (and we won't have the overwritten case)